core/src/test/java/io/jenkins/pluginhealth/scoring/probes/HasUnreleasedProductionChangesProbeTest.java

```diff
@@ -0,0 +1,179 @@
+                    assertThat(commitDate, greaterThan(pluginReleaseDate));
+                    assertThat(plugin)
+                        .usingRecursiveComparison()
+                        .comparingOnlyFields("releaseTimestamp")
+                        .isEqualTo(ProbeResult.error(HasUnreleasedProductionChangesProbe.KEY, ""));
+                    count++;
+                }
+                System.out.println("Had " + count + " commits");
+            }
+
+        }
+        final HasUnreleasedProductionChangesProbe probe = getSpy();
+        final ProbeResult result = probe.apply(plugin, ctx);
+
+
+    }
+
+    // check that a commit on pom.xml before the latest release is returning a SUCCESS.
+    @Test
+    void commitOnPomFileBeforeLatestReleaseDateShouldReturnSuccess() throws IOException, GitAPIException {
+        final Path repository = Files.createTempDirectory("test-foo-bar");
+
+        final Plugin plugin = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+
+        ZonedDateTime releaseTimestamp = ZonedDateTime.now();
+        when(plugin.getReleaseTimestamp()).thenReturn(releaseTimestamp);
+
+        when(plugin.getDetails()).thenReturn(Map.of(
+            SCMLinkValidationProbe.KEY, ProbeResult.success(SCMLinkValidationProbe.KEY, "")
+        ));
+        when(ctx.getScmRepository()).thenReturn(repository);
+
+        try(Git git = Git.init().setDirectory(repository.toFile()).call()) {
+
+            final Path pom = Files.createFile(repository.resolve("pom.xml"));
+
+            // creating commit
+
+            PersonIdent defaultCommitter = new PersonIdent(git.getRepository());
+            PersonIdent committer = new PersonIdent(defaultCommitter, Date.from(plugin.getReleaseTimestamp().minusDays(1).toInstant()));
+
+            git.add().addFilepattern("pom.xml").call();
+            git.commit().setMessage("Imports pom.xml file").setSign(false).setCommitter(committer).call();
+
+            final HasUnreleasedProductionChangesProbe probe = getSpy();
+
+            for (RevCommit commit : git.log().call()) {
+                long timestamp = commit.getCommitTime();
+                String dateString = plugin.getReleaseTimestamp().toString();
+
+                Instant timestampInstant = Instant.ofEpochSecond(timestamp);
+                DateTimeFormatter formatter = DateTimeFormatter.ISO_ZONED_DATE_TIME;
+                ZonedDateTime zonedDateTime = ZonedDateTime.parse(dateString, formatter);
+                Instant dateInstant = zonedDateTime.toInstant();
+
+                final ProbeResult result = probe.apply(plugin, ctx);
+                assertThat(commit.getFullMessage().equals("Imports pom.xml file"));
+                assert !timestampInstant.isBefore(dateInstant):result.failure("unreleased-production-changes", "unreleased-production-changes does not meet the criteria to be executed on null");
```

Comment:
2 observations:

1. FYI: this is the syntax being used on this line: `assert condition : message;`
where, "condition" is the expression that you want to check, and "message" is an optional message that will be displayed if the condition is false.

So in your case, the condition being checked is `!timestampInstant.isBefore(dateInstant)`. If this condition is false, a new ProbeResult object will be created (and returned) with the `id` as "unreleased-production-changes", `message` as "unreleased-production-changes does not meet the criteria to be executed on null", and `status` as "ResultStatus.FAILURE". And this is not what we want to do I think.

2. How did this string `"unreleased-production-changes does not meet the criteria to be executed on null"` come into the picture? I think you added it here because this is what you're currently seeing when you run the test with a debugger, is that right?

Reply:
> 1. where, "condition" is the expression that you want to check, and "message" is an optional message that will be displayed if the condition is false.

You are right about this. In the "message" section I was trying to check whether the method return failure or success.



> 2\. I think you added it here because this is what you're currently seeing when you run the test with a debugger, is that right?

Right. That is why I added it. 

I made a mistake. I should check for `success` instead of `failure`.

Reply:
The problem is, the status of the `result` variable is currently `ERROR` and it should be `SUCCESS`.
And it is of type `ERROR` because we didn't fulfill all the probe result requirements (mentioned under `getProbeResultRequirement()` of this probe).

Out of the 2 elements in the array:
```
public String[] getProbeResultRequirement() {
        return new String[]{SCMLinkValidationProbe.KEY, LastCommitDateProbe.KEY};
    }
```
we've only mocked 1st entry and that's why you're seeing that string.